packages/abi-coder/src/interface.test.ts

```diff
@@ -256,8 +257,52 @@ describe('Interface', () => {
       },
     ]);
 
-    expect(functionInterface.getFunction('entry_one').format()).toEqual(
+    expect(functionInterface.getFunction('entry_one').getInputsSignature()).toEqual(
       'entry_one(u64,s(bool,u64))'
     );
   });
+
+  it('can encode struct with dynamic typing', () => {
+    functionInterface = new Interface([
+      {
+        type: 'function',
+        name: 'entry_one',
+        inputs: [
+          {
+            name: 'my_u64',
+            type: 'u64',
+          },
+          {
+            name: 'my_struct',
+            type: 'struct MyStruct',
+            components: [
+              {
+                name: 'dummy_a',
+                type: 'bool',
+              },
+              {
+                name: 'dummy_b',
+                type: 'u64',
+              },
+            ],
+            typeArguments: [
+              {
+                name: 'T',
+                type: 'b256',
+              },
+              {
+                name: 'U',
+                type: 'bool',
+              },
+            ],
+          },
+        ],
+        outputs: [{ name: 'ret', type: 'u64' }],
+      },
+    ]);
+
+    expect(functionInterface.getFunction('entry_one').getInputsSignature()).toEqual(
+      'entry_one(u64,s<b256,bool>(bool,u64))'
```

Comment:
👍🏼

Reply:
nice that we have your PR with array/enum already merged. Fixes are included here, plus some more tests specifics to array/enum in dynamic typing structure.

Plus fix of #433  (which I didn't notice before when reviewing your PR)